pkg/healthcheck/healthcheck.go

```diff
@@ -2191,6 +2210,84 @@ func checkResources(resourceName string, objects []runtime.Object, expectedNames
 	return nil
 }
 
+// Check if there's a pod with the "opaque ports" annotation defined but a
+// service selecting the aforementioned pod doesn't define it
+func (hc *HealthChecker) checkMisconfiguredOpaquePortAnnotations(ctx context.Context, namespace string) error {
+	services, err := hc.GetServices(ctx)
+	if err != nil {
+		return err
+	}
+
+	var errStrings []string
+
+	for _, service := range services {
+		if service.Spec.ClusterIP == "None" {
+			// skip headless services; they're handled differently
+			continue
+		}
+
+		labelSet := labels.Set(service.Spec.Selector)
+		pods, err := hc.kubeAPI.CoreV1().Pods(namespace).List(
+			ctx,
+			metav1.ListOptions{LabelSelector: labelSet.AsSelector().String()},
+		)
+		if err != nil {
+			return err
+		}
+
+		if mismatch := misconfiguredOpaquePortAnnotationsInService(service, pods); mismatch != nil {
+			errStrings = append(
+				errStrings,
+				mismatch.Error(),
+			)
+		}
+	}
+
+	if len(errStrings) >= 1 {
+		return fmt.Errorf(strings.Join(errStrings, "\n"))
+	}
+
+	return nil
+}
+
+func misconfiguredOpaquePortAnnotationsInService(service corev1.Service, pods *corev1.PodList) error {
+	for _, pod := range pods.Items {
+		if err := misconfiguredOpaqueAnnotation(service, pod); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func misconfiguredOpaqueAnnotation(service corev1.Service, pod corev1.Pod) error {
+	svcAnnotations := service.Annotations
+	podAnnotations := pod.Annotations
+
+	// If the pod has the annotation, check that the service has it as well
+	if podAnnotation, found := podAnnotations[k8s.ProxyOpaquePortsAnnotation]; found {
+		if svcAnnotation, found := svcAnnotations[k8s.ProxyOpaquePortsAnnotation]; found {
+			if svcAnnotation != podAnnotation {
+				return fmt.Errorf("Pod %s and service %s have the annotation but values don't match", pod.Name, service.Name)
```

Comment:
I know these were my suggestions but can we change these to output similar to `kubectl get all` where it prefixes resources with their type? Same comment for the other errors we can return in this function!

```suggestion
				return fmt.Errorf("pod/%s and service/%s have the annotation but values don't match", pod.Name, service.Name)
```